### PR TITLE
Change flow companies

### PR DESCRIPTION
**Pull request by silviolrjunior**
*Change view index [admin/companies]
*Change view new [admin/company]
*Change view show/edit [company, admin/company]

screen shot 2015-08-27 at 4 47 17 pm

screen shot 2015-08-27 at 4 47 19 pm

screen shot 2015-08-27 at 4 47 23 pm

### CODE REVIEW

File: app/mirage/config.js [hunk 1]
```diff
@@ -8,6 +8,8 @@ export default function() {
   this.post('/companies');
   this.put('/companies/:id');
   this.delete('/companies/:id');
+  this.get('/company/:id');
```
Comment (fernandes) [inline app/mirage/config.js]: on rails to get a company is `/companies/:id` lets keep the rails restful routes pattern

File: app/mirage/factories/company.js [hunk 2]
```diff
@@ -1,12 +1,11 @@
-/*
-  This is an example factory definition.
-
-  Create more files in this directory to define additional factories.
-*/
 import Mirage, {faker} from 'ember-cli-mirage';
 
 export default Mirage.Factory.extend({
-  name: faker.company.companyName()
+  name: function() {
```
Comment (fernandes) [inline app/mirage/factories/company.js]: why needed to change to an anonymous function?

avoid anonymous function, when needed use:

```javascript
name: function generateCompanyName() {
  return faker.company.companyName();
},

File: app/routes/admin/companies/company.js [hunk 3]
```diff
@@ -2,15 +2,35 @@ import Ember from 'ember';
 
 export default Ember.Route.extend({
   model: function(params) {
-    return this.store.find('company', params.company_id);
+    return Ember.RSVP.hash({
```
Comment (fernandes) [inline app/routes/admin/companies/company.js]: JS promises? cool hehehhe

File: app/routes/company.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
+import Ember from 'ember';
+
+export default Ember.Route.extend({
+  model: function(params) {
+    return Ember.RSVP.hash({
+      company: this.store.find('company', params.company_id),
+      themes: this.store.findAll('theme')
+    });
+  },
+  serialize: function(company) {
```
Comment (fernandes) [inline app/routes/company.js]: curiosity: why this serialize?

File: tests/acceptance/companies-test.js [hunk 5]
```diff
@@ -41,12 +41,12 @@ test('edit a company', function(assert) {
     fillIn('input#name', 'Acme Inc');
     click('button#save');
     click("a:contains('Acme Inc')");
-    click("a:contains('Edit')");
+    click("button:contains('Change Name')");
     fillIn('input#name', 'New Company Name');
-    click('button#save');
+    click('button#save_bottom');
```
Comment (fernandes) [inline tests/acceptance/companies-test.js]: bottom? shouldn't be button?

File: app/mirage/factories/company.js (hunk 2 above)
Comment (silviolrjunior) [inline app/mirage/factories/company.js]: I just followed the example file created by mirage (contact.js)

Comment (fernandes) [inline app/mirage/factories/company.js]: got it... don't know if really need an anonymous function, but if for some reason you need, name the function so its easy to find on a backtrace

File: tests/acceptance/companies-test.js (hunk 5 above)
Comment (silviolrjunior) [inline tests/acceptance/companies-test.js]: It has two buttons to save for better usability, one on top and one below haha

Comment (fernandes) [inline tests/acceptance/companies-test.js]: ahhhh so its really bottom... cool :sunglasses: hahahah

**Pull request merged**